dev/benchmarks/microbenchmarks/lib/geometry/matrix_utils_transform_rect_bench.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import '../common.dart';
+
+const int _kNumIterations = 1000000;
+const int _kNumWarmUp = 10000;
+
+void main() {
+  assert(false, "Don't run benchmarks in checked mode! Use 'flutter run --release'.");
+  print('MatrixUtils.transformRect benchmark...');
+
+  Matrix4 _makePerspective(double radius, double angle, double perspective) {
+    return MatrixUtils.createCylindricalProjectionTransform(
+      radius: radius,
+      angle: angle,
+      perspective: perspective,
+    );
+  }
+
+  final List<Matrix4> _transforms = <Matrix4>[
+    Matrix4.identity()..scale(1.2, 1.3, 1.0)..rotateZ(0.1),
+    Matrix4.identity()..translate(12.0, 13.0, 10.0),
+    Matrix4.identity()..scale(1.2, 1.3, 1.0)..translate(12.0, 13.0, 10.0),
+  ];
+  final List<Matrix4> _perspectiveTransforms = <Matrix4>[
+    _makePerspective(10.0, math.pi / 8.0, 0.3),
+    _makePerspective( 8.0, math.pi / 8.0, 0.2),
+    _makePerspective( 1.0, math.pi / 4.0, 0.1)..rotateX(0.1),
+  ];
+  final List<Rect> _rects = <Rect>[
+    const Rect.fromLTRB(1.1, 1.2, 1.5, 1.8),
+    const Rect.fromLTRB(1.1, 1.2, 0.0, 1.0),
+    const Rect.fromLTRB(1.1, 1.2, 1.3, 1.0),
+    const Rect.fromLTRB(-1.1, -1.2, 0.0, 1.0),
+    const Rect.fromLTRB(-1.1, -1.2, -1.5, -1.8),
+  ];
+  final List<Offset> _offsets = <Offset>[
+    const Offset(1.1, 1.2),
+    const Offset(1.5, 1.8),
+    const Offset(0.0, 0.0),
+    const Offset(-1.1, -1.2),
+    const Offset(-1.5, -1.8),
+  ];
+  assert(_rects.length.gcd(_transforms.length) == 1);
```

Comment:
The expectation is that benchmarks are run with asserts off.  In fact, the assert up on line 17 should make all of these unreachable.

Make make them regular `if` and `throw`s?

Reply:
D'oh! I even ran into that in testing so I should have known better.

I switched it to a comment since the benchmark still runs fine, it's just "better" if we use all combinations of transforms and points/rects.